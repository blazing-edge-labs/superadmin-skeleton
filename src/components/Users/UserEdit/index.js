import React from 'react'
import {
  Edit,
  SimpleForm,
  DisabledInput,
  TextInput,
  DateField,
  required,
  email,
} from 'admin-on-rest'

const UserEdit = (props) => (
  <Edit {...props}>
    <SimpleForm>
      <DisabledInput source="id"/>
      <TextInput label="Email" source="email" validate={[required, email]}/>
      <DateField label="Created At" source="createdAt" showTime options={{ timeZoneName: 'short' }}/>
    </SimpleForm>
  </Edit>
)

export default UserEdit
