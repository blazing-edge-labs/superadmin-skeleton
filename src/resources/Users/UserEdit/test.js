import React from 'react'
import { shallow } from 'enzyme'

import UserEdit from './'

describe('UserEdit', () => {
  const userEdit = shallow(<UserEdit/>)

  it('renders properly', () => {
    expect(userEdit).toMatchSnapshot()
  })
})
