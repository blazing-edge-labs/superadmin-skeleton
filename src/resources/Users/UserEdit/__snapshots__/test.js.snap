// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserEdit renders properly 1`] = `
<Connect(getContext(Edit))>
  <Connect(ReduxForm)>
    <DisabledInput
      addField={true}
      source="id"
    />
    <TextInput
      addField={true}
      label="Email"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      options={Object {}}
      source="email"
      type="text"
      validate={
        Array [
          [Function],
          [Function],
        ]
      }
    />
    <pure(DateField)
      addLabel={true}
      label="Created At"
      options={
        Object {
          "timeZoneName": "short",
        }
      }
      showTime={true}
      source="createdAt"
    />
  </Connect(ReduxForm)>
</Connect(getContext(Edit))>
`;
